static/src/javascripts/projects/common/modules/commercial/adfree/maintain-adfree-freshness.js

```diff
@@ -0,0 +1,36 @@
+define([
+    'common/utils/cookies',
+    'common/utils/config',
+    'common/utils/storage',
+    'common/modules/identity/api',
+    'common/modules/commercial/adfree/renew-adfree-status'
+], function (
+    cookies,
+    config,
+    storage,
+    identity,
+    renewAdfreeStatus
+) {
+    return maintainAdfreeFreshness;
+
+    function maintainAdfreeFreshness() {
+        if (featureEnabled() && identity.isUserLoggedIn() && userNeedsNewAdfreeCookie()) {
+            renewAdfreeStatus.renew();
+        }
+    }
+
+    function userNeedsNewAdfreeCookie() {
+        var adfreeCookie = cookies.get('gu_adfree_user');
+        return (adfreeCookie === null) || adfreeCookieStale();
+    }
+
+    function adfreeCookieStale() {
+        var adfreeCookieExpiry = storage.local.get('gu.adfree.user.expiry'),
```

Comment:
I had thought about that. I know that dotcom prefer to use as few cookies as possible, for the sake of avoiding page bloat. But I agree that like should match like. Really I would prefer to handle both in local storage, but the service already returns a cookie. Perhaps I will migrate this value to the cookiejar instead.